crates/utils/src/esplora_client.rs

```diff
@@ -2,15 +2,16 @@ use bitcoin::{
     block::Header, consensus, hashes::hex::FromHex, BlockHash, CompactTarget, MerkleBlock, Network,
     Transaction, TxMerkleNode, Txid,
 };
-use eyre::Result;
+use eyre::{Error, Result};
 use reqwest::{Client, Url};
 use serde::Deserialize;
 use std::str::FromStr;
 use tracing::*;
 
-const ESPLORA_MAINNET_URL: &str = "https://blockstream.info/api/";
-const ESPLORA_TESTNET_URL: &str = "https://blockstream.info/testnet/api/";
+const ESPLORA_TESTNET_URL: &str = "https://btc-testnet.interlay.io";
+const ESPLORA_MAINNET_URL: &str = "https://btc-mainnet.interlay.io";
```

Comment:
The `tx/{tx_id}/hex` endpoint is not working on Blockstream's Esplora API (it times out), so I had to update the constant to use a different base URL.

Here's a eg for some Testnet transaction:

- **BOB Esplora**:  
  [`https://btc-testnet.interlay.io/tx/ffc0c6ffa168aada32b6ee012695c5c7097274570a130aa4aa5622a779a96f18/hex`](https://btc-testnet.interlay.io/tx/ffc0c6ffa168aada32b6ee012695c5c7097274570a130aa4aa5622a779a96f18/hex) 
- **Blockstream Esplora**:  
  [`https://blockstream.info/testnet/api/tx/ffc0c6ffa168aada32b6ee012695c5c7097274570a130aa4aa5622a779a96f18/hex`](https://blockstream.info/testnet/api/tx/ffc0c6ffa168aada32b6ee012695c5c7097274570a130aa4aa5622a779a96f18/hex)  (times out)